pkg/cmd/pr/create/create.go

```diff
@@ -659,64 +588,110 @@ func NewCreateContext(opts *CreateOptions) (*CreateContext, error) {
 		}
-		headBranchLabel = headBranch
-		isPushEnabled = true
-	} else if idx := strings.IndexRune(headBranch, ':'); idx >= 0 {
-		headBranch = headBranch[idx+1:]
 	}
 
-	gitClient := opts.GitClient
-	if ucc, err := gitClient.UncommittedChangeCount(context.Background()); err == nil && ucc > 0 {
-		fmt.Fprintf(opts.IO.ErrOut, "Warning: %s\n", text.Pluralize(ucc, "uncommitted change"))
+	targetHeadBranchConfig, err := gitClient.ReadBranchConfig(context.Background(), targetHeadBranch)
+	if err != nil {
+		return nil, err
 	}
 
-	var headRepo ghrepo.Interface
-	var headRemote *ghContext.Remote
+	// See if we can determine if this branch has been push previously with
+	// Git configurations and @{push} revision syntax.
+	remotePushDefault, err := opts.GitClient.RemotePushDefault(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// Suppressing these errors as we have other means of computing the PullRequestRefs when these fail.
+	parsedPushRevision, _ := opts.GitClient.ParsePushRevision(ctx, targetHeadBranch)
+	pushDefault, err := opts.GitClient.PushDefault(ctx)
+	if err != nil {
+		return nil, err
+	}
 
-	headBranchConfig, err := gitClient.ReadBranchConfig(context.Background(), headBranch)
+	prRefs, err := shared.ParsePRRefs(targetHeadBranch, targetHeadBranchConfig, parsedPushRevision, pushDefault, remotePushDefault, targetBaseRepo, remotes)
 	if err != nil {
 		return nil, err
 	}
-	if isPushEnabled {
-		// TODO: This doesn't respect the @{push} revision resolution or triagular workflows assembled with
-		// remote.pushDefault, or branch.<branchName>.pushremote config settings. The finder's ParsePRRefs
-		// may be able to replace this function entirely.
-		if trackingRef, found := tryDetermineTrackingRef(gitClient, remotes, headBranch, headBranchConfig); found {
-			isPushEnabled = false
-			if r, err := remotes.FindByName(trackingRef.remoteName); err == nil {
-				headRepo = r
-				headRemote = r
-				headBranchLabel = trackingRef.branchName
-				if !ghrepo.IsSame(baseRepo, headRepo) {
-					headBranchLabel = fmt.Sprintf("%s:%s", headRepo.RepoOwner(), trackingRef.branchName)
+
+	// If the --head provided contains <repo_name>:<branch>  syntax, we need to use
+	// the provided owner instead of the owner of the base repository.
+	if targetHeadRepoOwner != "" {
+		prRefs.HeadRepo = ghrepo.New(targetHeadRepoOwner, prRefs.HeadRepo.RepoName())
+	}
+
+	var headRemote *ghContext.Remote
+
+	// We received the head repository and branch from ParsePRRefs, or inferred
+	// it from --head input, but we need to check if it's up-to-date with
+	// our local branch state.
+	// If it is, we can use it as the head repo for the PR
+	// and avoid prompting the user.
+	if prRefs.HeadRepo != nil && prRefs.BranchName != "" {
+		// Check if the head branch is up-to-date with the local branch
+		headRemote, err := remotes.FindByRepo(prRefs.HeadRepo.RepoOwner(), prRefs.HeadRepo.RepoName())
+		if err == nil {
```

Comment:
What happens if `err != nil`?

Reply:
If there is an `err` here, then something happened when finding the remote, and we can't use that remote. This causes `promptForHeadRepo` to never get set to `false` and so we'd prompt later for the head repo, which gives us the head remote also.

Reply:
I think you're describing the logic that follows this. My question was more about what happens to the error. That definitely wasn't clear!

Looks like it gets swallowed (I was wondering if you knew of handling of it further down). As such, I would suggest this change (including the above line)

```suggestion
                headRemote, _ := remotes.FindByRepo(prRefs.HeadRepo.RepoOwner(), prRefs.HeadRepo.RepoName())
		if headRemote != nil {
```

That way we are clearly communicating that we don't care about this error and reall just care about the `headRemote`

Reply:
I wonder if we do care about this error though - doesn't an error indicate that we shouldn't try and use that remote? Without reading the implementation of `FindByRepo`, I'm not sure what an error means, and I'm not sure if we should couple it that way anyway.

I'll think on this more 🤔

Reply:
Fixed in 1b3e2018f236a5594551ac981dd99245f5a18109

Also re-ran ATs:

```
--- PASS: TestPullRequests (0.00s)
    --- PASS: TestPullRequests/pr-create-respects-simple-pushdefault (8.87s)
    --- PASS: TestPullRequests/pr-checkout-by-number (9.03s)
    --- PASS: TestPullRequests/pr-create-from-manual-merge-base (9.04s)
    --- PASS: TestPullRequests/pr-comment (9.09s)
    --- PASS: TestPullRequests/pr-create-with-metadata (9.85s)
    --- PASS: TestPullRequests/pr-view-same-org-fork (11.53s)
    --- PASS: TestPullRequests/pr-create-basic (7.63s)
    --- PASS: TestPullRequests/pr-create-respects-branch-pushremote (16.82s)
    --- PASS: TestPullRequests/pr-create-respects-remote-pushdefault (16.97s)
    --- PASS: TestPullRequests/pr-create-respects-push-destination (17.12s)
    --- PASS: TestPullRequests/pr-view (7.45s)
    --- PASS: TestPullRequests/pr-create-respects-remote-colon-branch-syntax (18.66s)
    --- PASS: TestPullRequests/pr-view-status-respects-simple-pushdefault (8.35s)
    --- PASS: TestPullRequests/pr-create-from-issue-develop-base (11.54s)
    --- PASS: TestPullRequests/pr-checkout (13.91s)
    --- PASS: TestPullRequests/pr-view-outside-repo (7.09s)
    --- PASS: TestPullRequests/pr-view-status-respects-remote-pushdefault (16.69s)
    --- PASS: TestPullRequests/pr-list (6.78s)
    --- PASS: TestPullRequests/pr-view-status-respects-push-destination (10.31s)
    --- PASS: TestPullRequests/pr-merge-merge-strategy (11.35s)
    --- PASS: TestPullRequests/pr-create-without-upstream-config (7.77s)
    --- PASS: TestPullRequests/pr-checkout-with-url-from-fork (12.49s)
    --- PASS: TestPullRequests/pr-merge-rebase-strategy (12.52s)
    --- PASS: TestPullRequests/pr-view-status-respects-branch-pushremote (16.13s)
PASS
ok      github.com/cli/cli/v2/acceptance        35.382s
```